proposals/0389-dynamic-min-deposit.md

```diff
@@ -0,0 +1,249 @@
+
+Let target growth per slot be `G_target = 2 GiB/epoch / slots_per_epoch`.
+
+Compute error: `e = G_slot - G_target` (positive if growth exceeds target).
+
+Integrate with asymmetric bounds:
+
+```
+I_next = clamp( I + e, I_min = -4_000_000_000, I_max = +2_000_000_000 )
+```
+
+Deadband and discrete asymmetric steps:
+
+```
+if I_next >= H:                    # above-band (sustained over-target)
+    min_deposit_next = min_deposit_current * 1.10   # +10%
+elif I_next <= -H:                 # below-band (sustained under-target)
+    min_deposit_next = min_deposit_current * 0.95   # -5%
+else:
+    min_deposit_next = min_deposit_current          # hold
+```
+
+Notes:
+
+- No proportional term; the controller is integral-only with a deadband. The
+  goal is to keep price pinned at the floor (0.1x) under normal conditions and
+  react only to pronounced spikes.
+- `H` (deadband threshold) is a protocol constant (see Protocol constants).
+- `I` is the integral accumulator over error; large positive values indicate
+  sustained overshoot; large negative values indicate sustained slack.
+
+Rationale: the main goal of the controller is to supervise state growth and
+engage effectively and predictably when the safety threshold is violated.
+
+- Asymmetric updates: bias towards upwards adjustments after engagement is
+  triggered allows a faster response to excessive state growth and a more
+  gradual return to baseline when conditions normalize.
+- Asymmetric integral bounds: the absolute value of `I_min` being larger than
+  `I_max` allows for the accumulation of a buffer to prevent smaller spikes
+  from causing a `min_deposit` adjustment, thereby increasing stability of the
+  0.1x floor price. A return to baseline can happen faster after a high
+  allocation event has passed, with the lower `I_max` value.
+- Deadband: avoids oscillations around the safety threshold.
+
+**Clamps:**
+
+Apply clamps after the update to ensure `min_deposit` stays within bounds:
+
+```
+min_deposit_next = clamp(
+  0.1 * min_balance_legacy,
+  1.0 * min_balance_legacy,
+  min_deposit_next
+)
+```
+
+where `min_balance_legacy` is the fixed minimum-balance constant used prior to
+this feature activation.
+
+### Sysvar exposure
+
+Expose `min_deposit` via the existing Rent sysvar to avoid introducing a new
+sysvar and to preserve compatibility with existing on-chain programs.
+
+The Rent sysvar account is `SysvarRent111111111111111111111111111111111`.
+
+### Persistence and snapshot integrity
+
+- The two per-slot deltas (`data_size_delta`, `num_accounts_delta`) and the
+  integral accumulator `I` MUST be backed up into dedicated sysvars so they are
+  covered by snapshot integrity checks via the accounts lthash. This avoids the
+  need to include these values in the bank hash.
+- On startup, validators MUST load these values exclusively from the sysvars,
+  rather than any non-account serialized fields in the snapshot, ensuring that
+  integrity-checked values are used.
```

Comment:
The new fields aren't going into the snapshot (as there is no SIMD/proposal to do so). Thus may not need to mention otherwise, as to me it implied this could be possible.